### PR TITLE
feat/addBrokenLinksReporter

### PR DESCRIPTION
**Pull request by kishore03109**
## Problem

Closes [insert issue #]

## Solution

**Breaking Changes**

- [ ] Yes - this PR contains breaking changes
  - Details ...
- [ ] No - this PR is backwards compatible with ALL of the following feature flags in this [doc](https://www.notion.so/opengov/Existing-feature-flags-518ad2cdc325420893a105e88c432be5)

**Features**:

- Details ...

**Improvements**:

- Details ...

**Bug Fixes**:

- Details ...

## Before & After Screenshots

**BEFORE**:

**AFTER**:

## Tests

- [ ] Unit tests (using `npm run tests`)
- [ ] e2e tests (comment on this PR with the text `!run e2e`)
- [ ] Smoke tests

## Deploy Notes

**New environment variables**:

- `env var` : env var details

**New scripts**:

- `script` : script details

**New dependencies**:

- `dependency` : dependency details

**New dev dependencies**:

- `dependency` : dependency details

### CODE REVIEW

**Comment by kishore03109** [comment]
Current dependencies on/for this PR:
* **#1811** <a href="https://app.graphite.dev/github/pr/isomerpages/isomercms-frontend/1811?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#1810** <a href="https://app.graphite.dev/github/pr/isomerpages/isomercms-frontend/1810?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `develop`

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/isomerpages/isomercms-frontend/1810?utm_source=stack-comment).

File: src/layouts/LinkReport/LinksReport.tsx [hunk 1]
```diff
@@ -0,0 +1,342 @@
...
+
+export const LinksReportBanner = () => {
+  const { siteName } = useParams<{ siteName: string }>()
+  const { mutate: refreshLinkChecker } = useRefreshLinkChecker(siteName)
+  const onClick = () => {
+    refreshLinkChecker(siteName)
+  }
+
+  const { data: brokenLinks } = useGetBrokenLinks(siteName)
+
```
Comment (alexanderleegs) [inline src/layouts/LinkReport/LinksReport.tsx:63]: Hmm by default, does a site have no errors if the error checker isn't run prior? e.g. for new sites

File: src/layouts/LinkReport/LinksReport.tsx [hunk 2]
```diff
@@ -0,0 +1,342 @@
...
+  links: NonPermalinkErrorDto[]
+}) => {
+  // can use any link since we know all the links are from the same page
+  const { viewablePageInStaging, viewablePageInCms } = links[0]
+  const { siteName } = useParams<{ siteName: string }>()
+  const { data: stagingUrl, isLoading: isStagingUrlLoading } = useGetStagingUrl(
+    siteName
+  )
+
+  const viewableLinkInStaging = stagingUrl + viewablePageInStaging.slice(1) // rm the leading `/`
```
[the remark this answers is not in the document]
Comment (kishore03109) [reply src/layouts/LinkReport/LinksReport.tsx:105]: hey you are right, caught this in testing + commit is [here](https://github.com/isomerpages/isomercms-frontend/pull/1811/files#diff-0fa359897de6fb099ee581e42b7a541530ceeacf8cfc32d721cb6fc83cac991dR92) 

tldr this is fixed upstream

File: src/layouts/LinkReport/LinksReport.tsx (hunk 1 above)
Comment (kishore03109) [reply src/layouts/LinkReport/LinksReport.tsx:63]: hey you are right on this, there is no auto link checker, since it triggered manually  
Not doing this for this experiment since the ones that we are experimenting with are a select -hardcoded list, which we can control.
in the future if doesnt exist, be should create! 
even if we forget, this pr already has a recovery path as shown here: 

https://github.com/isomerpages/isomercms-frontend/assets/42832651/929c93dd-28aa-4730-b3f9-c8d8a0695d53

File: src/layouts/LinkReport/LinksReport.tsx [hunk 3]
```diff
@@ -0,0 +1,342 @@
...
+    .filter((_, index) => index % 2 === 0)
+    .slice(2)
+    .join(" / ")
+    .replace(/-/g, " ")
+  if (breadcrumb.endsWith(".md")) {
+    breadcrumb = breadcrumb.slice(0, -3)
+  }
+
+  return breadcrumb
+}
```
Comment (alexanderleegs) [inline src/layouts/LinkReport/LinksReport.tsx:53]: Hmm does the broken link checker also support homepage/navbar/contact us? The breadcrumb wouldn't work properly for those right

File: src/layouts/LinkReport/LinksReport.tsx (hunk 1 above)
Comment (kishore03109) [reply src/layouts/LinkReport/LinksReport.tsx:63]: @alexanderleegs wah you caught me tripping off alarms. 

[2578d9a](https://github.com/isomerpages/isomercms-frontend/pull/1811/commits/2578d9a97dc0b45057d2a206ab71831e3300447d)
Fixing this issue upstream so that fe dont not ping be in the event of non feature flagged

File: src/layouts/LinkReport/LinksReport.tsx (hunk 3 above)
Comment (kishore03109) [reply src/layouts/LinkReport/LinksReport.tsx]: arh shucks you are right, missed these out. [8837e07](https://github.com/isomerpages/isomercms-frontend/pull/1810/commits/8837e079b04d8958509ecaf6f3945927f45ff1fb) as a todo.

navbar is not affected since its a yml, currently the scanner doesnt scan. 
bug exists for homepage + contact us, will fix upstream

**Review by alexanderleegs — APPROVED** [review]
Approving with followups to do since this is feature flagged

**Comment by kishore03109** [comment]
### Merge activity

* **Feb 22, 3:03 AM EST**: @kishore03109 started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/isomerpages/isomercms-frontend/1810).
